recipes-bsp/u-boot/files/0024-arm-dts-iot2050-Use-external-blob-file-for-OTP-updat.patch

```diff
@@ -0,0 +1,79 @@
+From 92d1ec191fa6d961e9c8d8898f2f40aeca0f78d8 Mon Sep 17 00:00:00 2001
+From: Baocheng Su <baocheng.su@siemens.com>
+Date: Thu, 12 May 2022 22:00:43 +0800
+Subject: [PATCH 24/24] arm: dts: iot2050: Use external blob file for OTP
+ update command
+
+Use external blob otpcmd.bin to replace the 0xff filled OTP update
+command block.
+
+This otpcmd.bin is generated from the customer keys.
+
+This is preparing for the secure boot on IOT2050. User can build a
+firmware image that contains the otpcmd data which will be programmed
+into the extend OTP area on next booting by following the instructions
+on device.
```

Comment:
Why do we need this here? We don't technically need such a block for building the firmware. We only need it when provisioning something.

I'm missing the user story here.

Reply:
Yep, this is a optional block just as the wdt firmware

Reply:
This is for the one-shot firmware of both otp programming and signed u-boot booting.

Or I could remodel it to a seperate script to `dd` the otpcmd.bin to the block, and append it to dpkg_build.

The current way looks more nature to me, just as the wdt story.

Reply:
The wdt firmware is less optional than this block because it is signed along with the other critical assets. But we should likely document it as we have it enabled now default as I just realized.

In contrast, the otpcmd block is signed separately and can also be deployed separately.